core/include/stdout_redirect.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Part of esp-open-rtos
+ * Copyright (C) 2016 Oto Petrik <oto.petrik@gmail.com>
+ * BSD Licensed as described in the file LICENSE
+ */
+
+#ifndef _STDOUT_REDIRECT_H_
+#define _STDOUT_REDIRECT_H_
```

Comment:
This header uses `_reent` structure so it's better to include `<sys/reent.h>` here.